File: 2053/solution.cpp
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        std::unordered_map<std::string, int> valuesMap;
        for (int i = 0; i < arr.size(); i++) {
            if (!valuesMap.count(arr[i])) {
                valuesMap[arr[i]] = 1;
            } else {
                valuesMap[arr[i]]++;
            }
        }
        for (int i = 0; i < arr.size(); i++) {
            if (valuesMap[arr[i]] == 1) {
                k -= 1;
            }
            if (k == 0) {
                return arr[i];
            }
        }
        return "";
    }
};

File: 2053/solution.js
/**
 * @param {string[]} arr
 * @param {number} k
 * @return {string}
 */
var kthDistinct = function(arr, k) {
  let distinctStrings = []
  let indices = []
  arr.forEach((str, i) => {
      let idx = arr.indexOf(str);
      while (idx !== -1) {
          indices.push(idx);
          idx = arr.indexOf(str, idx + 1);
      }
      if (indices.length === 1) distinctStrings.push(str)

      indices = []
  })
  return distinctStrings[k-1] ? distinctStrings[k-1] : ""
};